src/main/scala/beam/sim/population/PopulationAdjustment.scala

```diff
@@ -56,6 +56,12 @@ trait PopulationAdjustment extends LazyLogging {
     populationWithAttributes
   }
 
+  // required for testing purposes
+  def logInfo(message: String): Unit = logger.info(message)
+
+  // required for testing purposes
+  def logError(message: String): Unit = logger.error(message)
```

Comment:
Basically we can use it. Though it looks a little strange. What we can do is to create a custom Logback appender (https://www.baeldung.com/custom-logback-appender) and in logback-test.xml add it to the PopulationAdjustment only. Then we can get the appender programmatically and cleanup/check it before/after running a test.